src/estimation/duplicates_check.py

```diff
@@ -0,0 +1,62 @@
+import pandas as pd
+import logging
+import sys
+
+DupLogger = logging.getLogger(__name__)
+
+def count_unique(df: pd.DataFrame) -> pd.DataFrame:
+    '''Returns dataframe with count column
+    giving count of unique row
+
+    Args:
+        df (pd.DataFrame): Pandas dataframe
+        with responses data
+
+    Returns:
+        duplicates(pd.DataFrame): Pandas dataframe
+        with duplicate rows or empty dataframe 
+        if no duplicates exist 
+    '''
+
+    # group by all columns and
+    # add col "count" with count of unique rows
+    duplicates = df.groupby(
+        df.columns.tolist()
+        ).size().reset_index().rename(columns={0:'count'})
+
+    return duplicates
+
+
+def duplicates_check(df: pd.DataFrame) -> pd.DataFrame:
+    '''Creates error message with duplicates
+    if they exist. If not gives an info message
+
+    Args:
+        df (pd.DataFrame): Pandas dataframe
+        with responses data
+    '''
+
+    # obtain counts of unique rows
+    # and filter for any not 1
+    duplicates = count_unique(df)
+    duplicates = duplicates[duplicates["count"] != 1]
+    
+    if len(duplicates) > 0:
+        DupLogger.error(
+            (
+                "The following dataframe shows duplicate rows "
+                "(see the count column for a count of duplicate rows)"
+                f": {duplicates}"
+
+            )
+        )
+
+        #exit pipeline if any duplicates
+        sys.exit()
```

Comment:
I'm wondering if you need to do sys.exit() ? I think the pipeline  will stop anyway when you do an error to the logger (that's what we've done elsewhere)

Reply:
Instead of doing the sys.exit, can you raise an error so it's consistent with the rest of the pipeline?
eg
        raise ValueError